segments/vcs.p9k

```diff
@@ -122,41 +122,24 @@ p9k::set_default P9K_VCS_HIDE_TAGS false
 p9k::set_default P9K_VCS_INTERNAL_HASH_LENGTH "8" # Default: Just display the first 8 characters of our changeset-ID.
 p9k::set_default P9K_VCS_DIR_SHORTEN_DELIMITER $'\U2026'
 p9k::set_default P9K_VCS_SHOW_SUBMODULE_DIRTY true
-
-################################################################
-# Register segment helper default values
-p9k::set_default P9K_VCS_SHOW_SUBMODULE_DIRTY false
 p9k::set_default P9K_VCS_GIT_ALWAYS_SHOW_REMOTE_BRANCH false
 
 function +vi-git-untracked() {
-  [[ -z "${vcs_comm[gitdir]}" || "${vcs_comm[gitdir]}" == "." ]] && return
-
-  # If we are in a .git folder, do not check for untracked files.
-  [[ "${PWD:A}" =~ "\.git/" ]] && return
-
-  # If we are in a repos root folder, vcs_comm[gitdir] yields ".git".
-  # Inside the .git dir itself (and not a subdir of it) the variable
-  # yields ".". In any other case (either a subdirectory of .git or
-  # the repo itself), the value of vcs_comm[gitdir] is the absolute
-  # path to the .git directory.
-  # Therefore we can step up a directory, if we are inside the .git
-  # folder. And in any other case, use the parent directory of the
-  # gitdir.
-  local repoDir="."
-  # Getting the parent dir of the current dir "." is still ".", so
-  # is is safe to do this always.
-  [[ "${vcs_comm[gitdir]}" != ".git" ]] && repoDir="${vcs_comm[gitdir]:A:h}"
-  [[ "${vcs_comm[gitdir]}" == "." ]] && repoDir="${PWD:A:h}"
-
-  if [[ "$P9K_VCS_SHOW_SUBMODULE_DIRTY" == "true" && "$(command git submodule foreach --quiet --recursive 'command git ls-files --others --exclude-standard')" != "" ]]; then
-    hook_com[unstaged]+=" ${__P9K_ICONS[VCS_UNTRACKED]}"
-    VCS_WORKDIR_HALF_DIRTY=true
-  elif [[ "$(command git ls-files --others --exclude-standard "${repoDir}")" != "" ]]; then
-    hook_com[unstaged]+=" ${__P9K_ICONS[VCS_UNTRACKED]}"
-    VCS_WORKDIR_HALF_DIRTY=true
-  else
-    VCS_WORKDIR_HALF_DIRTY=false
+  # get the root for the current repo or submodule
+  local repoTopLevel="$(git rev-parse --show-toplevel 2> /dev/null)"
+  # dump out if we're outside a git repository (which includes being in the .git folder)
+  [[ $? != 0 || -z $repoTopLevel ]] && return
+
+  local untrackedFiles=$(command git ls-files --others --exclude-standard "${repoTopLevel}")
+
+  if [[ -z $untrackedFiles && "$P9K_VCS_SHOW_SUBMODULE_DIRTY" != "false" ]]; then
```

Comment:
I agree... it definitely improves readability and I’ll go ahead and commit this change.  Let me walk you through my thinking on this, though, because I’m still a little bothered by a bigger question here.  Is there a convention we want to follow for settings that use a string for a boolean value?

Requiring an exact match certainly works, as you suggest, and I did have it that way initially.  I changed it because I followed the following thinking:

- If the user sets a value themselves, they probably don’t want the default behavior
- Since this is a boolean setting, there’s only one other behavior
- If they’ve set *any* value, assume that's the one they want.
- The default value is `"false"`, so if the value is any different (e.g., `"1"`, `"on"`, `"yes"`, `"enable"`), assume they want `"true"`.  *(Note: this was erroneous, I know... originally, both settings were used in the file, and I assumed that we’d opt for faster processing by default.)*

We might also consider having a function to handle this, which could return truth by actually evaluating the string against a set of true-ish and false-ish values (0/1, true/false, yes/no, on/off, enable(d)/disable(d)).

Reply:
I already did my suggested changes directly in your branch. Sorry for fiddling in your branch, but I want to get this change out asap.

About boolean values and comparing them: 
Hmm. Indeed I wrap boolean values in quotes by a force of habit. And it is a good idea to have a dedicated function that checks for multipe true-ish values. But this is out of scope for this PR.

`P9K_VCS_SHOW_SUBMODULE_DIRTY` defaulted to `true` not long ago. I flipped it because of performance reasons. But I am on the verge to let it default to `true` again, because this matches the default value of the past releases (only `false` in current `master` and `next`).

Reply:
> I already did my suggested changes directly in your branch. Sorry for fiddling in your branch, but I want to get this change out asap.

No worries!  Appreciate it.

> About boolean values and comparing them:
> Hmm. Indeed I wrap boolean values in quotes by a force of habit. And it is a good idea to have a dedicated function that checks for multipe true-ish values. But this is out of scope for this PR.

Agreed... just food for thought.
 
> `P9K_VCS_SHOW_SUBMODULE_DIRTY` defaulted to `true` not long ago. I flipped it because of performance reasons. But I am on the verge to let it default to `true` again, because this matches the default value of the past releases (only `false` in current `master` and `next`).

Sure.  I’m personally fine with it either way, but it does slow things down noticeably if you’re working in a project like powerlevel9k or prezto, and you’ve got a lot of sub-sub-submodules, especially on the 2014 MBP that I use at work.

Reply:
> Sure. I’m personally fine with it either way, but it does slow things down noticeably if you’re working in a project like powerlevel9k or prezto, and you’ve got a lot of sub-sub-submodules, especially on the 2014 MBP that I use at work.

How bad is it? I am interested to gather some statistics about that. Could you redo the tests from https://github.com/bhilburn/powerlevel9k/pull/1046#issuecomment-435549640 on your machine?